framework/include/base/AuxGroupExecuteMooseObjectWarehouse.h

```diff
@@ -107,7 +107,9 @@ AuxGroupExecuteMooseObjectWarehouse<T>::updateDependObjects(
     }
 
     std::uint16_t remaining_flags = already_added ? not_initial_flag_mask : all_flags;
-    if (depend_aux.find(object_ptr->name()) != depend_aux.end() ||
+    if ((object_ptr->isParamValid("force_preaux") &&
+         object_ptr->template getParam<bool>("force_preaux")) ||
```

Comment:
You don't need the template keyword here. Just makes it confusing.

Reply:
it didn't compile for me without.

Reply:
of course it is needed here